sklearn/metrics/tests/test_score_objects.py

```diff
@@ -136,37 +141,118 @@ def test_all_scorers_repr():
         repr(scorer)
 
 
-def test_check_scoring():
-    # Test all branches of check_scoring
+def check_scoring_validator_for_single_metric_usecases(scoring_validator):
+    # Test all branches of single metric usecases
     estimator = EstimatorWithoutFit()
     pattern = (r"estimator should be an estimator implementing 'fit' method,"
                r" .* was passed")
-    assert_raises_regexp(TypeError, pattern, check_scoring, estimator)
+    assert_raises_regexp(TypeError, pattern, scoring_validator, estimator)
 
     estimator = EstimatorWithFitAndScore()
     estimator.fit([[1]], [1])
-    scorer = check_scoring(estimator)
+    scorer = scoring_validator(estimator)
     assert_true(scorer is _passthrough_scorer)
     assert_almost_equal(scorer(estimator, [[1]], [1]), 1.0)
 
     estimator = EstimatorWithFitAndPredict()
     estimator.fit([[1]], [1])
     pattern = (r"If no scoring is specified, the estimator passed should have"
                r" a 'score' method\. The estimator .* does not\.")
-    assert_raises_regexp(TypeError, pattern, check_scoring, estimator)
+    assert_raises_regexp(TypeError, pattern, scoring_validator, estimator)
 
-    scorer = check_scoring(estimator, "accuracy")
+    scorer = scoring_validator(estimator, "accuracy")
     assert_almost_equal(scorer(estimator, [[1]], [1]), 1.0)
 
     estimator = EstimatorWithFit()
-    scorer = check_scoring(estimator, "accuracy")
+    scorer = scoring_validator(estimator, "accuracy")
     assert_true(isinstance(scorer, _PredictScorer))
 
     estimator = EstimatorWithFit()
-    scorer = check_scoring(estimator, allow_none=True)
+    scorer = scoring_validator(estimator, allow_none=True)
     assert_true(scorer is None)
 
 
+def check_multimetric_scoring_single_metric_wrapper(*args, **kwargs):
+    scorers, is_multi = check_multimetric_scoring(*args, **kwargs)
+    # For all single metric use cases, it should register as not multimetric
+    assert_false(is_multi)
+    if scorers is not None:
+        scorers = list(scorers.values())
+        assert_equal(len(scorers), 1)
+        scorers = scorers[0]
+    return scorers
+
+
+def test_check_scoring_and_check_multimetric_scoring():
+    check_scoring_validator_for_single_metric_usecases(check_scoring)
+    # To make sure the check_scoring is correctly applied to the constituent
+    # scorers
+    check_scoring_validator_for_single_metric_usecases(
+        check_multimetric_scoring_single_metric_wrapper)
+
+    # For multiple metric use cases
+
+    # Make sure it works for the valid cases
+
+    for scoring in (('accuracy', 'precision'), ['precision', 'accuracy'],
+                    {'accuracy': make_scorer(accuracy_score),
+                     'precision': make_scorer(precision_score)}):
+        estimator = EstimatorWithFitAndPredict()
+        estimator.fit([[1]], [1])
+
+        scorers, is_multi = check_multimetric_scoring(estimator, scoring)
+        assert_true(is_multi)
```

Comment:
somewhere you also need to test that `check_multimetric_scoring(['accuracy'])[1] is True` and `check_multimetric_scoring({'acc': 'accuracy'})[1] is True`